### PR TITLE
docs($animate): detail options object

### PR DESCRIPTION
**Pull request by wesleycho**
- Document options object used in various $animate methods

This should address #12645.

### CODE REVIEW

File: src/ng/animate.js [hunk 1]
```diff
@@ -420,7 +420,14 @@ var $AnimateProvider = ['$provide', function($provide) {
...
        * @param {DOMElement=} after the sibling element after which the element will be appended
-       * @param {object=} options an optional collection of options/styles that will be applied to the element
+       * @param {object=} options an optional collection of options/styles that will be applied to the element.
+       *   The object has the following properties:
+       *
+       *   - **addClass** - `{string}` - space-separated CSS classes to add to element
+       *   - **from** - `{Object}` - CSS properties & values from beginning of animation
+       *   - **removeClass** - `{string}` - space-separated CSS classes to remove from element
+       *   - **to** - `{Object}` - CSS properties & values at end of animation
+       *
```
Comment (gkalpak) [inline src/ng/animate.js]: Could you also document the `options` argument of other methods (e.g. `move()`) ?

**Comment by gkalpak** [comment]
This should indeed fix #12645. Feel free to add `Fixes #12645` in the commit message.

File: src/ng/animate.js [hunk 2]
```diff
@@ -420,7 +420,13 @@ var $AnimateProvider = ['$provide', function($provide) {
        * @param {DOMElement} parent the parent element which will append the element as
        *   a child (so long as the after element is not present)
        * @param {DOMElement=} after the sibling element after which the element will be appended
-       * @param {object=} options an optional collection of options/styles that will be applied to the element
+       * @param {object=} options an optional collection of options/styles that will be applied to the element.
+       *   The object has the following properties:
+       *
+       *   - **addClass** - `{string}` - space-separated CSS classes to add to element
+       *   - **from** - `{Object}` - CSS properties & values from beginning of animation
```
Comment (Narretz) [inline src/ng/animate.js]: "from the beginning" -> "at the beginning"

File: src/ng/animate.js [hunk 3]
```diff
@@ -420,7 +420,13 @@ var $AnimateProvider = ['$provide', function($provide) {
        * @param {DOMElement} parent the parent element which will append the element as
        *   a child (so long as the after element is not present)
        * @param {DOMElement=} after the sibling element after which the element will be appended
-       * @param {object=} options an optional collection of options/styles that will be applied to the element
+       * @param {object=} options an optional collection of options/styles that will be applied to the element.
+       *   The object has the following properties:
```
Comment (Narretz) [inline src/ng/animate.js]: It can have the following properties, but it doesn't have to. The only requirement is that "from" must have a matching "to"

**Pull request merged**